### DISCUSSION TITLE
How to synthesize a three-dimensional pore network model？

### DISCUSSION THREAD

**Discussion by Devin-one** [PMEAL/porespy#878] [Q&A]
Hello Professor. I have a series of two-dimensional slice images. I can extract a two-dimensional pore network model through Snow2. But if I want to synthesize a three-dimensional pore network model, what should I do? Thank you in advance for your reply.

**Comment by jgostick** [comment]
You'll need a 3D image.  Are the 'series' of 2D images you have taken from a 3D tomogram, or are you referring to microscope type images?  If the former, then you can put them into a 3D stack and pass that to snow2 (with the necessary pre-processing).  If the latter, then you're pretty much out of luck.  There are algorithms out there for generating 3D images from the 2D one, but they are pretty advanced.

**Comment by Devin-one** [comment]
I think my images would fall into the former category you mentioned. I sliced a cube model consecutively. My idea is to synthesize these 2D sliced images into a 3D pore network model. I tried to do this with 3 sliced images but failed. The following image contains my code and its output. Is it because I have too few slice images? Since each image is 500 x 500 in size, can I make the 3D image also 500 pixels high by controlling the sequence spacing of these 3 images.
<img width="604" alt="image" src="https://github.com/PMEAL/porespy/assets/93040662/f726f276-ad43-4953-ab67-67f05bb74fdd">

**Reply by jgostick** [reply]
This is not really an OpenPNM/PoreSpy problem, and I don't have the time to do tech support and training for people.  There even if you got this working, only have 3 slices in the z-direction is not enough.  It should probably be more like 500x500x500.  Perhaps the 2 "General" tutorials [here](https://porespy.org/_examples/index.html#general) would be helpful for you?

Also, the thresholding of that image above is not great, as much of the inner section of the grains is missing.
